cmd/quadlet/main.go

```diff
@@ -210,6 +211,66 @@ func enableServiceFile(outputPath string, service *parser.UnitFile) {
 	}
 }
 
+func isSha256Digest(imageName string) bool {
+	if strings.HasPrefix(imageName, "sha256:") {
+		return true
+	}
+
+	// only fully specified digests are matched
+	if len(imageName) != 64 {
+		return false
+	}
+
+	for _, c := range imageName {
+		if !unicode.Is(unicode.Hex_Digit, c) {
+			return false
+		}
+	}
+
+	return true
+}
+
+func isFullyQualifiedName(imageName string) bool {
+	if isSha256Digest(imageName) {
+		return true
+	}
+
+	firstSlash := strings.Index(imageName, "/")
+	if firstSlash == -1 {
+		// No domain or path, not fully qualified
+		return false
+	}
+
+	// What is before the first slash can be a domain or a path
+	domain := imageName[:firstSlash]
+
+	// If its a a domain (has dot or port or is "localhost") it is considered fq
+	if strings.ContainsAny(domain, ".:") || domain == "localhost" {
```

Comment:
(For the record, note that this can be also a non-DNS IP address. IPv4 on all versions, in the future possibly also IPv6. Luckily the `".:"` check matches both of these cases.)